assertj-error-prone/src/main/java/com/palantir/assertj/errorprone/PreferAssertj.java

```diff
@@ -414,6 +428,28 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
                 return describeMatch(tree);
             }
         }
+        if (ASSERT_THROWS.matches(tree, state)) {
+            String replacement = String.format(
+                    "assertThatThrownBy(%s).isInstanceOf(%s)", argSource(tree, state, 1), argSource(tree, state, 0));
```

Comment:
This technically isn't an exact replacement, a JUnit5 `Executable` may not be a AssertJ `ThrowingCallable`. But in practice these are arguments are almost certainly going to be lambdas, so this seems reasonable.

Reply:
Should we check that the argument is a lambda tree, otherwise just describe a match without a suggested fix?

Reply:
In cases where we're not confident I think it's safer to complain loudly and fail rather than to produce code that won't compile. Might be helpful to add test coverage for such a case using either an anonymous Executable implementation or a passed value of that type.

Reply:
I'd also be fine just removing the `ASSERT_THROWS` and `ASSERT_DOES_NOT_THROW` replacements.

Does it makes sense for the fall-through return statement to return a match instead of `NO_MATCH`? We've already done the `FAST_CHECK` above so we know that we're using one of the legacy classes.

Reply:
Done